### DISCUSSION TITLE
Modified Jyers Ender 3 v2 firmware on Ender 6?

### DISCUSSION THREAD

**Discussion by sassysander** [Jyers/Marlin#1034] [Ideas]
Just a thought I had and wanted to get a sanity check on. I own a few Ender 3 v2s and an Ender 6 and am running the excellent Jyers firmware on the Ender 3 v2s. Wouldn't it be great to have the same firmware on the 6?

So I was thinking, since the v4.3.1 mainboard in the Ender 6 is pretty much identical to the v4.2.2/4.2.7, why not buy and install the Ender 3 v2 stock LCD to the Ender 6 and run a modified version of the Jyers firmware?

All that's left to do is check motor direction and fiddle around in Marlin via a serial connection to set up the specifics of the printer (corexy, acceleration, jerk, etc.) and we should be good to go yes? Or am I missing something?

**Comment by smooreace** [comment]
In theory it should work... what happens if you plug in a v2 screen to your
Ender 6 right now? Im kinda thinking you might have to do some pin work in
the firmware too. But a quick look at the schematic, or the screen plugs
pin-out,
 for the 4.3.1 board will help determine that

**Reply by Reggi3** [reply]
#define BOARD_INFO_NAME      "Creality v4.3.1"
#define DEFAULT_MACHINE_NAME "Creality3D"
//
// Steppers
//
#define X_STEP_PIN                        PB8
#define X_DIR_PIN                         PB7
#define Y_STEP_PIN                        PC2
#define Y_DIR_PIN                         PB9
#include "pins_CREALITY_V4.h"

Looks like the only pin differences are the X and Y step/dir pins.

Next up is to check the configs for 4.3.1 board and see which defines are available but in theory, it looks like it's the same pinout for the lcd as it is in other 4.2.x boards.
